src/fuzzfetch/fetch.py

```diff
@@ -335,6 +335,124 @@ def _revision_paths(cls, rev, branch, target_platform):
+                                      ', '.join(Fetcher.TEST_CHOICES)))
+        test_group.add_argument('--full-symbols', action='store_true',
+                                help='Download the full crashreport-symbols.zip archive.')
+
+        misc_group = self.parser.add_argument_group('Misc. Arguments')
+        misc_group.add_argument('-n', '--name',
+                                help='Specify a name (default=auto)')
+        misc_group.add_argument('-o', '--out', default=os.getcwd(),
+                                help='Specify output directory (default=.)')
+        misc_group.add_argument('--dry-run', action='store_true',
+                                help="Search for build and output metadata only, don't download anything.")
+
+        near_group = self.parser.add_argument_group('Near Arguments',
+                                                    "If the specified build isn't found, iterate over " +
+                                                    "builds in the specified direction")
+        near_args = near_group.add_mutually_exclusive_group()
+        near_args.add_argument('--nearest-newer', action='store_const', const=Fetcher.BUILD_ORDER_ASC, dest='nearest',
+                               help="Search from specified build in ascending order")
+        near_args.add_argument('--nearest-older', action='store_const', const=Fetcher.BUILD_ORDER_DESC, dest='nearest',
+                               help="Search from the specified build in descending order")
+
+    def sanity_check(self, args):
+        """
+        Perform parser checks
+
+        @type arg: list
+        @param arg: a list of arguments
+        """
+        if re.match(r'(\d{4}-\d{2}-\d{2}|[0-9A-Fa-f]{12}|[0-9A-Fa-f]{40}|latest)$', args.build) is None:
+            # this is a custom build
+            # ensure conflicting options are not set
+            if args.branch is not None:
+                self.parser.error('Cannot specify --build namespace and branch argument: %s' % args.branch)
+            if args.debug:
+                self.parser.error('Cannot specify --build namespace and --debug')
+            if args.asan:
+                self.parser.error('Cannot specify --build namespace and --asan')
+            if args.tsan:
+                self.parser.error('Cannot specify --build namespace and --tsan')
+            if args.fuzzing:
+                self.parser.error('Cannot specify --build namespace and --fuzzing')
+            if args.coverage:
+                self.parser.error('Cannot specify --build namespace and --coverage')
+            if args.valgrind:
+                self.parser.error('Cannot specify --build namespace and --valgrind')
+
+    def parse_args(self, argv=None):
+        """
+        Parse and validate args
+
+        @type argv: list
+        @param argv: a list of arguments
+        """
+        args = self.parser.parse_args(argv)
+        self.sanity_check(args)
```

Comment:
Part 3/5 of suggestion for comment line 956.
```suggestion
        self.build_is_ns = re.match(r'(\d{4}-\d{2}-\d{2}|[0-9A-Fa-f]{12}|[0-9A-Fa-f]{40}|latest)$', args.build) is None
        self.sanity_check(args)
```